src/mem/alloc.h

```diff
@@ -507,33 +515,39 @@ namespace snmalloc
       {
         // This is a large alloc redirect.
         ss = ss - (1ULL << (size - 64));
-        size = global_pagemap.get((void*)ss);
+        size = global_pagemap.get(ss);
       }
 
       if (size == 0)
       {
         if constexpr ((location == End) || (location == OnePastEnd))
           // We don't know the End, so return MAX_PTR
-          return (void*)-1;
+          return (uintptr_t)~0;
```

Comment:
`reinterpret_cast`, though what you probably actually mean is `uintptr_t(~0ULL)`.

Reply:
So I am not liking the `ULL`, after getting it wrong for 32bit pointers. 

I am thinking about adding an `constexpr UINTPTR_MAX` to bits to encapsulate the few places we want -1 as a pointer in one place.